### PR TITLE
feat: `bring github`

### PR DESCRIPTION
**Pull request by eladcon**

### CODE REVIEW

File: github/README.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# github
+
```
Comment (eladb) [inline github/README.md:2]: Please add some context and description

File: github/README.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+`sh
+npm i @winglibs/github
+`
+
+## Usage
+
+`js
+bring github;
+
+let adder = new github.Adder();
```
Comment (eladb) [inline github/README.md:18]: Please update with documentation

File: github/lib.test.w [hunk 3]
```diff
@@ -0,0 +1,76 @@
+bring fs;
+bring expect;
+bring util;
+bring "./probot/probot.w" as probot;
+let key = fs.readFile("/path/to/private-key.pem");
+
+let handler = inflight (ctx) => {
+  fs.writeFile("/tmp/111", "pull requesdt opened");
```
Comment (eladb) [inline github/lib.test.w:8]: Probably better to use mkdtemp to put this is a unique temp dir

File: github/lib.w [hunk 4]
```diff
@@ -0,0 +1,7 @@
+bring "./probot/probot.w" as probot;
+
+pub class ProbotApp extends probot.ProbotApp {
```
Comment (eladb) [inline github/lib.w:3]: I am wondering if this needs library needs to be called `probot` instead of GitHub.

Comment (eladb) [inline github/lib.w:3]: What's the reason we have this class?

File: github/octokit/types.w [hunk 5]
```diff
@@ -0,0 +1,311 @@
...
+	inflight getCommit(options: GetCommitProps): GetCommitResponse;
+	inflight compareCommits(options: CompareCommitsProps): CompareCommitsResponse;
+	inflight createOrUpdateFileContents(options: CreateOrUpdateFileContentsProps): BaseResponse;
+	inflight createForAuthenticatedUser(options: CreateRepoProps): BaseResponse;
+	inflight createInOrg(options: CreateOrgRepoProps): BaseResponse;
+	inflight delete(options: DeleteRepoProps): BaseResponse;
+	inflight listForAuthenticatedUser(options: ListForAuthenticatedUserProps): ListReposResponse;
+	inflight listForOrg(options: ListForOrgProps): ListReposResponse;
+}
+
```
Comment (eladb) [inline github/octokit/types.w:304]: Is this generated somehow or manually written?

File: github/probot/types.w [hunk 6]
```diff
@@ -0,0 +1,86 @@
+bring "../octokit/types.w" as octokit;
+
+struct IProbotRepositoryOwner {
```
Comment (eladb) [inline github/probot/types.w:3]: This is a data type, not an interface, so the I prefix is not needed

```suggestion
struct ProbotRepositoryOwner {
```

Comment (eladb) [reply github/probot/types.w:3]: Also below...

**Review by eladb — CHANGES_REQUESTED** [review]

File: github/lib.test.w (hunk 3 above)
Comment (ekeren) [reply github/lib.test.w:8]: It is a debug entry, we will remove it

File: github/lib.test.w [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+
+  // find all changed mdfiles by comparing the commits of the PR
+  let compare = ctx.octokit.repos.compareCommits(
+    owner: repo.owner.login,
+    repo: repo.name,
+    base: ctx.payload.pull_request.base.sha,
+    head: ctx.payload.pull_request.head.sha,
+  );
+
+  fs.writeFile("/tmp/112", "{Json.stringify(compare.data.commits)}");
```
Comment (ekeren) [inline github/lib.test.w:19]: Remove

File: github/lib.w (hunk 4 above)
Comment (eladb) [reply github/lib.w:3]: Okay.

I guess the readme will be useful so I can understand how this is supposed to be used.

File: github/lib.test.w (hunk 3 above)
Comment (ekeren) [inline github/lib.test.w:8]:
```suggestion
```

File: github/lib.test.w (hunk 7 above)
Comment (ekeren) [inline github/lib.test.w:19]:
```suggestion
```

**Review by ekeren — APPROVED** [review]

**Pull request merged**